go/stream/message.go

```diff
@@ -20,6 +20,9 @@ const (
 	mapKeySourceJobRunID  = "sourceJobRunID"
 	mapKeySourceTaskRunID = "sourceTaskRunID"
 	mapKeyTraceID         = "traceID"
+	mapKeySourceType      = "sourceType"
+	mapKeyReason          = "reason"
```

Comment:
Wouldn't we populate this field for all messages? I was under the impression that some of these might not apply to all type of messages, correct? If that is the case, perhaps we should discuss about introducing a message type (as in given a message type we can expect/read a different `struct`).

Reply:
Yeah, SourceType can be sent for all but Reason and Stage are specific to webhook or proc errors.

Reply:
Started a thread: https://rudderlabs.slack.com/archives/C067J99V4AU/p1718784519698499